staff/views.py

```diff
@@ -267,6 +271,84 @@ def make_announcement(request):
     return render(request, "make_announcement.html", {"form": form})
 
 
+@group_required("staff")
+def post_form(request):
+    if request.method == "POST":
+        form = PostFormForm(request.POST, request.FILES)
+        if form.is_valid():
+            posted_form = Form(
+                name=form.cleaned_data.get("name"),
+                description=form.cleaned_data.get("description"),
+                form=request.FILES["form"],
+                created_by=DjangoUser.objects.get(id=request.user.id),
+            )
+            if form.cleaned_data.get("esign") is not None:
+                posted_form.esign = form.cleaned_data.get("esign")
+            posted_form.save()
+            posted_form.recipient_groups.set(form.cleaned_data.get("recipient_groups"))
+            posted_form.recipient_classrooms.set(
+                form.cleaned_data.get("recipient_classrooms")
+            )
+            messages.add_message(request, messages.SUCCESS, "Successfully Posted Form")
+            if form.cleaned_data.get("email_recipients"):
+                data = request.POST.copy()
+                email_list = get_emails_from_form(data)
+                email_posted_form(request, form, email_list)
+            distribute_forms(request, posted_form, form)
+            return redirect("staff")
+        else:
+            messages.error(
+                request,
+                "Form NOT made, your form form was not valid, please try again.",
+            )
+            return redirect("staff")
```

Comment:
We might want a fall-through instead of a redirect on failure, I think the redirect may clear any data that the user entered into the fields?

Reply:
This seems to be the answer we are looking for, this could be implemented in many places throughout the app (if so, we should create an issue):  

https://stackoverflow.com/questions/14647723/django-forms-if-not-valid-show-form-with-error-message/14647770

Reply:
That looks right

Reply:
Issue created